### PR TITLE
introduce a release note for release

### PR DESCRIPTION
**Pull request by leseb**
From now on, let's get into the habit of editing the
`CHANGELOG.md` file on PR adding notable features and/or
breaking changes that users need to be aware of.

Signed-off-by: Sébastien Han <seb@redhat.com>

### CODE REVIEW

**Comment by nathan-weinberg** [comment]
Maybe we should put this in the `docs/` driectory - wdyt @leseb?

**Comment by alimaredia** [comment]
@leseb In the scenario where someone opens up a PR with a breaking change. Should the author edit this file in that PR or will a release manager/maintainer be responsible for updating this doc before the release.

**Comment by leseb** [comment]
> @leseb In the scenario where someone opens up a PR with a breaking change. Should the author edit this file in that PR or will a release manager/maintainer be responsible for updating this doc before the release.

From my perspective, whoever introduces feature X with a PR has to update the PendingReleaseNotes.md in that PR. Most of the time, maintainers will have to ask for it but we could do better and add a note in the PR template.

**Comment by leseb** [comment]
> Maybe we should put this in the `docs/` directory - wdyt @leseb?

In most of the projects I've worked with, we had this file at the root of the repo. It's simple, easy to find and won't cause any trouble if `docs` disappears or something else.

**Comment by alimaredia** [comment]
> > @leseb In the scenario where someone opens up a PR with a breaking change. Should the author edit this file in that PR or will a release manager/maintainer be responsible for updating this doc before the release.
> 
> From my perspective, whoever introduces feature X with a PR has to update the PendingReleaseNotes.md in that PR. Most of the time, maintainers will have to ask for it but we could do better and add a note in the PR template.

That sounds good to me. What do you think about adding automation around that to this PR? Say a maintainer adds a "Release Notes" label, this then the PR can't get merged unless the new file changed?

**Comment by leseb** [comment]
> I support this in general. It's probably worth an update to the release process doc for how to handle this, as well.

Thanks for the suggestion, I added a new section in `docs/release-strategy.md`. PTAL. Thanks!

**Review by nathan-weinberg — APPROVED** [review]

File: docs/release-strategy.md [hunk 1]
```diff
@@ -68,3 +68,29 @@ The following are the steps for how Y-stream and Z-stream releases gets cut.
...
+need to update the `PendingReleaseNotes.md` file. Clearly describe the changes, their impact, and
+any actions users might need to take. We want clear, concise, and user-friendly notes.
+
+### Branching for a New Release
+
+Each time we prepare for a new major (X) release, we branch out from the main codebase.
+As part of this branching process, the contents of `PendingReleaseNotes.md` are reviewed and
+finalized.
+
+### Resetting the Release Notes
```
Comment (alimaredia) [inline docs/release-strategy.md:90]: @leseb why not just get rid of the overhead of clearing `PendingReleaseNotes.md` and have the file being a running list of breaking changes with those for the upcoming release at the top. This was how it was done in Ceph (https://github.com/ceph/ceph/blob/main/PendingReleaseNotes). 

Though this file can get long I think there's value to having all of the breaking changes between versions in one place.

File: PendingReleaseNotes.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# v0.17 Pending Release Notes
+
+## Breaking Changes
+
+## Features
```
Comment (russellb) [inline PendingReleaseNotes.md:5]: This format will be necessary to allow for multiple release's worth of notes in the file.

```suggestion
# Release Notes

## v0.17 Pending Release Notes

### Breaking Changes

### Features
```

File: docs/release-strategy.md (hunk 1 above)
Comment (russellb) [reply docs/release-strategy.md:90]: +1

and in that case, I'd rename the file to `ReleaseNotes.md`, though `RELEASE_NOTES.md` would be more consistent with other files in the same directory.

Comment (leseb) [reply docs/release-strategy.md:90]: After some more digging, it looks like most of the notable open-source projects out there use a `CHANGELOG.md` file. The projects maintain a single `CHANGELOG.md` file that documents all releases. I like the suggestion @alimaredia @russellb, I think this approach provides a great balance between consistency, ease of management, and centralized information. Thanks!

File: docs/release-strategy.md [hunk 3]
```diff
@@ -68,3 +68,22 @@ The following are the steps for how Y-stream and Z-stream releases gets cut.
 1. Announce release via the following:
     - The `#announce` channel on Slack
     - The `announce` mailing list
+
+## Release notes
```
Comment (nathan-weinberg) [inline docs/release-strategy.md:72]:
```suggestion
## Release Notes
```

File: docs/release-strategy.md [hunk 4]
```diff
@@ -68,3 +68,22 @@ The following are the steps for how Y-stream and Z-stream releases gets cut.
...
+
+The project maintains a single `CHANGELOG.md` file that documents all releases.
+To ensure our users are well-informed about new features, improvements, and breaking changes, we
+maintain a `CHANGELOG.md` file. This file serves as a centralized place to document changes
+that will be included in the next major (X) release.
```
Comment (nathan-weinberg) [inline docs/release-strategy.md:77]:
```suggestion
that will be included in the next major (X) release.
```
Y-streams as well?

File: docs/release-strategy.md [hunk 5]
```diff
@@ -68,3 +68,22 @@ The following are the steps for how Y-stream and Z-stream releases gets cut.
...
+
+### Editing Release Notes
+
+When submitting a Pull Request (PR) that introduces notable features or breaking changes, committers
+need to update the `CHANGELOG.md` file. Clearly describe the changes, their impact, and
+any actions users might need to take. We want clear, concise, and user-friendly notes.
+
+### Branching for a New Release
+
+Each time we prepare for a new major (X) release, we branch out from the main codebase.
```
Comment (nathan-weinberg) [inline docs/release-strategy.md:87]: Same question here about Y-Streams

File: docs/release-strategy.md (hunk 4 above)
Comment (leseb) [reply docs/release-strategy.md:77]: The Y-streams are backports only right? So they will only include critical bug fixes, no features nor breaking changes. From my perspective, it's just Major release updates.

File: docs/release-strategy.md (hunk 5 above)
Comment (leseb) [reply docs/release-strategy.md:87]: https://github.com/instructlab/instructlab/pull/1231#discussion_r1626088030

File: docs/release-strategy.md (hunk 4 above)
Comment (nathan-weinberg) [reply docs/release-strategy.md:77]: Z-Streams are backports only

File: docs/release-strategy.md (hunk 5 above)
Comment (nathan-weinberg) [reply docs/release-strategy.md:87]: Responded there!

File: docs/release-strategy.md (hunk 4 above)
Comment (leseb) [reply docs/release-strategy.md:77]: Rah sorry, actually I mixed things up. I was talking about (Y) releases all along and not (X). In the end, both X and Y will get the same treatment. Thanks for pointing this out!

Comment (nathan-weinberg) [reply docs/release-strategy.md]: Np at all!

**Review by nathan-weinberg — APPROVED** [review]

**Review by russellb — APPROVED** [review]

**Pull request merged**